roles/metrics-export.lua

```diff
@@ -258,25 +298,47 @@ local function apply_http(conf)
     local enabled = {}
     for _, node in ipairs(conf) do
         if #(node.endpoints or {}) > 0 then
-            local host, port, err = parse_listen(node.listen)
-            if err ~= nil then
-                error("failed to parse URI: " .. err, 2)
+            local host, port, target
+            if node.server ~= nil then
+                local server = httpd_role.get_server(node.server)
+                host, port = server.host, server.port
+                target = node.server
+            elseif node.listen ~= nil then
+                target = node.listen
+                local err
+                host, port, err = parse_listen(node.listen)
+                if err ~= nil then
+                    error("failed to parse URI: " .. err, 2)
+                end
+            else
+                local server = httpd_role.get_server()
+                host, port = server.host, server.port
+                target = httpd_role.DEFAULT_SERVER_NAME
             end
-            local listen = node.listen
 
             http_servers = http_servers or {}
-            enabled[listen] = true
+            enabled[target] = true
+
+            if http_servers[target] == nil then
+                local httpd
+                local is_httpd_role = false
+                if node.listen ~= nil then
+                    httpd = http_server.new(host, port)
+                    httpd:start()
+                else
+                    httpd = httpd_role.get_server(target)
+                    is_httpd_role = true
+                end
 
-            if http_servers[listen] == nil then
-                local httpd = http_server.new(host, port)
-                httpd:start()
-                http_servers[listen] = {
+                http_servers[target] = {
                     httpd = httpd,
                     routes = {},
+                    is_httpd_role = is_httpd_role,
                 }
             end
-            local server = http_servers[listen]
-            local httpd = server.httpd
+
+            local server = http_servers[target]
+            local httpd = server.httpd or server
```

Comment:
Why do we need the change?
```suggestion
            local httpd = server.httpd
```

Reply:
The `httpd` role already returns an instance of httpd, so this declaration is required if `httpd` functionality is used.

Reply:
Fixed.